examples/rn-connection-and-error/frontend/app/hooks/useDemoSyncTriggers.ts

```diff
@@ -155,38 +170,43 @@ export function useDemoSyncTriggers() {
   }, [currentUser]);
 
   /**
-   * Pause and resume the session in order to apply new rules to schemas.
+   * Pause and resume the session in order to apply new rules to collections.
+   *
+   * @note
+   * This function is meant to be called after having first triggered a store change
+   * followed by a refresh of user data if an automatic session refresh does not occur.
+   * The store change will update permissions on the backend, making it necessary
+   * to refresh the session in order to see the effects in the UI.
    */
-  const refreshSession = useCallback(async () => {
-    logger.info('Pausing Session...');
-    realm.syncSession?.pause();
+  const refreshSession = useCallback(() => {
+    // To refresh the session, we first pause the session, causing sync to get
+    // disconnected. Once disconnected, the `handleConnectionChange()` listener
+    // will be called, wherein we manually resume the session if `reconnectAfterDisconnect`
+    // is set to `true`. This pattern can be used to react to connection changes.
+    reconnectAfterDisconnect = true;
+    disconnect();
```

Comment:
Figured it may be good to show users how to leverage the pattern of reacting to connection state changes via the connection listener. This way users won't have to have their own implementations of checking the connection in intervals, and it could be more reliable as well. (The connection listener callback will also take care of logging the new connection state.)

To better see this refactor, you can check out [this commit](https://github.com/realm/realm-js/commit/3935cdd94ee910c17d42502fe2815fefc608e688).